yasnippet.el

```diff
@@ -1083,7 +1083,7 @@ Has the following fields:
         (maphash #'(lambda (k v)
                      (let ((template (gethash name v)))
                        (when (and template
-                                  (eq uuid (yas--template-uuid template)))
+                                  (equal uuid (yas--template-uuid template)))
```

Comment:
No, I'm looking at just `intern`'in the keys. Like `(intern uuid yas--templates)` , where `yas--templates` is an `obarray`. And then `get` and `put` to store stuff in the symbol itself. It should really simplify all this stuff.